packages/origin-247-certificate/src/offchain-certificate/synchronize/handlers/persist.handler.ts

```diff
@@ -0,0 +1,32 @@
+import { SynchronizableEvent } from '../../repositories/CertificateEvent/CertificateEvent.repository';
+import { CertificateEventEntity } from '../../repositories/CertificateEvent/CertificateEvent.entity';
+import { CertificateCommandEntity } from '../../repositories/CertificateCommand/CertificateCommand.entity';
+import { ClaimPersistHandler } from './claim-persist.handler';
+import { IssuancePersistHandler } from './issuance-persist.handler';
+import { TransferPersistHandler } from './transfer-persist.handler';
+
+export interface PersistHandler {
+    canHandle(event: SynchronizableEvent): boolean;
+
+    handle(event: CertificateEventEntity, command: CertificateCommandEntity | null): Promise<void>;
+}
+
+export class PersistProcessor {
+    constructor(
+        private readonly claimPersistHandler: ClaimPersistHandler,
+        private readonly issuancePersistHandler: IssuancePersistHandler,
+        private readonly transferPersistHandler: TransferPersistHandler
+    ) {}
+
+    public async handle(event: SynchronizableEvent, command: CertificateCommandEntity | null) {
```

Comment:
why command can be null?

Reply:
As we may not find command corresponding to `event.commandId`. Should we just assume it always exists and use `!` operator?

Reply:
So how it works exactly? Why are we looking for command here? I don't understand.

Reply:
fixed